Dockerfile.prod

```diff
@@ -6,22 +6,39 @@ WORKDIR /app
 COPY . .
 WORKDIR /app/frontend/star-tides/
 RUN npm i --no-verbose
-RUN npm i -g @angular/cli@latest
+RUN npm i --no-verbose -g @angular/cli@latest
 RUN ng build \
     --configuration=production \
     --no-verbose \
+    --progress=false \
     --deploy-url='/'
 WORKDIR /
 
 FROM python:3.7-stretch as python_builder
 WORKDIR /app
 COPY . .
 COPY --from=node_builder /app/frontend/star-tides/dist/star-tides /app/static
-RUN pip3 install -r star_tides/requirements.txt
-CMD python3 run.py
+RUN pip3 install -q -r star_tides/requirements.txt
+# CMD python3 run.py
+# TODO(ljr): I think this CMD above is now a no-op?
+# We would like to have the Python code running, 
+# but ultimately start up the Nginx reverse proxy as well.
 
 FROM nginx:latest as nginx_builder
 COPY . .
 COPY --from=node_builder /app/frontend/star-tides/dist/star-tides /app/static
 COPY --from=python_builder /app /app
-COPY ./nginx.conf /etc/nginx/nginx.conf
\ No newline at end of file
+COPY ./nginx.conf /etc/nginx/nginx.conf
+# ENTRYPOINT [ "python3", "run.py" ]
+# CMD python3 run.py
+#
+# TODO(ljr): I don't think these ENTRYPOINTs or CMDs work since this is in the nginx phase/step of a
+# multi-step build and not the Python phase/step.
+# Perhaps the app isn't starting?
+
+FROM python_builder
+WORKDIR /app
+CMD python3 run.py
+# TODO(ljr): I think since this step/command is last,
+# Flask is running with werkzeug and not with the nginx reverse proxy.
```

Comment:
Werkzeug is a wsgi server for dev because it's a crappy app server. uwsgi and gunicorn are better wsgi app servers. Nginx can handle the onslaught of requests and filter out the crap/static content/etc, and then reverse proxy everything else to the app. Because of above, I think it'll be a lot less of a headache for us to separate nginx out from cloudrun and have whatever the load balancer/reverse proxy managed service is handle that. 

That way, just our app server runs in the api, then the bundled SPA stuff can sit on the webserver. 

Also, gcloud run instances shut down after not being used, right? Wouldn't that make it pretty expensive timewise to cold start because we'd have to boot the webserver and the app server together?

Reply:
I know we initially decided on this route, but I can't wrap my head around the benefit of it and how it would work b/c of the proxy and the app being in the same container. The load balancer would but kinda useless in this setting imo because everything is going to be balanced to one app server, instead of one load balancer being mapped to many app servers.